### PR TITLE
ONL-5035 Adding of indeterminate to the checkbox

### PR DESCRIPTION
**Pull request by derrick-g**
Added indeterminate to the checkbox, where the checkbox will have a horizontal line when one of the child check-boxes is selected, this will then become a check mark only if all child items are selected.

### CODE REVIEW

File: src/components/ec-checkbox/ec-checkbox.vue [hunk 1]
```diff
@@ -129,6 +139,22 @@ export default {
       },
     },
   },
+  watch: {
+    indeterminate(newValue) {
+      this.updateIndeterminate(newValue);
+    },
+  },
+  mounted() {
+    this.updateIndeterminate(this.indeterminate);
```
Comment (jafcalvente) [inline src/components/ec-checkbox/ec-checkbox.vue:148]: This is being called here in order to check if we have to mark as indeterminate the checkbox as soon as the component is mounted, right? Then maybe we could try to use the `immediate` option in the watch function instead:
```
watch: {
  indeterminate: {
    immediate: true,
    handler(newValue) {
      this.updateIndeterminate(newValue);
    },
  },
},
```

**Review by jafcalvente — APPROVED** [review]

File: src/components/ec-checkbox/ec-checkbox.vue [hunk 2]
```diff
@@ -30,14 +30,20 @@
           'ec-checkbox__check-icon-wrapper--focused': inputIsFocused,
           'ec-checkbox__check-icon-wrapper--checked': inputModel,
           'ec-checkbox__check-icon-wrapper--checked-and-focused': inputModel && inputIsFocused,
+          'ec-checkbox__check-icon-wrapper--indeterminate': indeterminate,
+          'ec-checkbox__check-icon-wrapper--indeterminate-and-focused': indeterminate && inputIsFocused,
```
Comment (jeger-at) [inline src/components/ec-checkbox/ec-checkbox.vue:34]: We also need a indeterminate-and-disabled style which should grey out the indeterminate state when disabled

File: src/components/ec-checkbox/ec-checkbox.vue [hunk 3]
```diff
@@ -30,14 +30,20 @@
...
           'ec-checkbox__check-icon-wrapper--error': isInvalid && !inputModel,
           'ec-checkbox__check-icon-wrapper--disabled': disabled,
           'ec-checkbox__check-icon-wrapper--checked-and-disabled': disabled && inputModel,
         }"
         @click="$refs.checkboxInput.click()"
       >
+        <span
+          v-if="indeterminate"
+          class="ec-checkbox__indeterminate-icon"
```
Comment (stavros-tomas) [inline src/components/ec-checkbox/ec-checkbox.vue:43]: I have my thoughts if this should be another icon and swap between icons instead. If no one else mention this you can go ahead and ignore :)

File: src/components/ec-checkbox/ec-checkbox.vue (hunk 1 above)
Comment (mcibique) [reply src/components/ec-checkbox/ec-checkbox.vue:148]: `updateIndeterminate` does use `$refs` which are only available after component has been mounted. `immediate` watchers are triggered before mounted, so the input's ref is still null and it won't get set.

File: src/components/ec-checkbox/ec-checkbox.vue (hunk 2 above)
Comment (derrick-g) [reply src/components/ec-checkbox/ec-checkbox.vue:34]: Good thinking, I will include those as well.

File: src/components/ec-checkbox/ec-checkbox.vue (hunk 3 above)
Comment (mcibique) [reply src/components/ec-checkbox/ec-checkbox.vue:43]: The icon for indeterminate is just a white line 10px by 2px. I think it is quite waste of resources trying to extract it into an icon. I don't see the icon to be reusable so it's okay to have it hard coded like this.

**Review by stavros-tomas — APPROVED** [review]

**Review by jeger-at — APPROVED** [review]

**Pull request merged**